### ISSUE TITLE
Broken help flags (-h --help)

### ISSUE THREAD

**Issue by tg** [codegangsta/cli#231]
The latest patch (854c52192c709766187afe5891a755bc97f82c38) breaks the `-h` `--help` flags when default action is set. Instead of help being shown, action is run. Example from the README:

```go
package main

import (
        "github.com/codegangsta/cli"
        "os"
)

func main() {
        app := cli.NewApp()
        app.Name = "boom"
        app.Usage = "make an explosive entrance"
        app.Action = func(c *cli.Context) {
                println("boom! I say!")
        }

        app.Run(os.Args)
}
```

When run:
```
$ ./cli -h
boom! I say!
```

Both `cli h` and `cli help` work as expected.

**Comment by tg** [comment]
The same holds for `-v` and `--version`.

**Comment by jszwedko** [comment]
Thank you for the report -- I didn't even consider that that PR would have that effect. Should be fixed by #232 (with additional test cases).

**Comment by jszwedko** [comment]
Closing this since #232 is merged, please let me know if this is still an issue.
